src/bpmncwpverify/builder/cwp_builder.py

```diff
@@ -70,35 +70,40 @@ def _add_and_check_expressions(
                 edge.parent_id = parent_id_ref
 
     def build(self) -> Result[Cwp, Error]:
-        expression_checker = ExpressionListener(self.symbol_table)
-        self._parse_states(self.states)
-        self._parse_edges(self.edges)
-
-        # This step assigns expressions to each edge and checks to make sure expression is valid
-        self._add_and_check_expressions(self.all_items, expression_checker)
-
-        end_states = [
-            state for state in self._cwp.states.values() if not state.out_edges
-        ]
-
-        self._cwp.start_states = {
-            id: state for id, state in self._cwp.states.items() if not state.in_edges
-        }
-
-        if not (end_states and self._cwp.start_states):
-            raise Exception("No start states or no end states")
-
-        self._cwp.states = {
-            id: state
-            for id, state in self._cwp.states.items()
-            if id not in self._cwp.start_states
-        }
-
-        # This step ensures connectivity of the graph and sets leaf edges
-        visitor = CwpConnectivityVisitor()
-        self._cwp.accept(visitor)
-
-        return Success(self._cwp)
+        try:
+            expression_checker = ExpressionListener(self.symbol_table)
+            self._parse_states(self.states)
+            self._parse_edges(self.edges)
+
+            # This step assigns expressions to each edge and checks to make sure expression is valid
+            self._add_and_check_expressions(self.all_items, expression_checker)
+
+            end_states = [
+                state for state in self._cwp.states.values() if not state.out_edges
+            ]
+
+            self._cwp.start_states = {
+                id: state
+                for id, state in self._cwp.states.items()
+                if not state.in_edges
+            }
+
+            if not (end_states and self._cwp.start_states):
+                raise Exception("No start states or no end states")
+
+            self._cwp.states = {
+                id: state
+                for id, state in self._cwp.states.items()
+                if id not in self._cwp.start_states
+            }
+
+            # This step ensures connectivity of the graph and sets leaf edges
+            visitor = CwpConnectivityVisitor()
+            self._cwp.accept(visitor)
+
+            return Success(self._cwp)
+        except Exception as e:
+            return Failure(e.args[0])
```

Comment:
updated this in the cwp_builder as well. Maybe should have put this in another PR